File: pkg/pwrgrep/testdata/fixtures/go/side-effects.go
package fixture

import (
	"database/sql"
	"net/http"
	"os"
	"os/exec"
)

func f(db *sql.DB) {
	// ruleid: go-side-effects
	exec.Command("ls").Run()
	// ruleid: go-side-effects
	os.WriteFile("out.txt", nil, 0o644)
	// ruleid: go-side-effects
	os.RemoveAll("tmp")
	// ruleid: go-side-effects
	http.Get("https://example.com")
	// ruleid: go-side-effects
	db.Query("SELECT 1")
	// ok: go-side-effects
	os.Getenv("HOME")
}

File: pkg/pwrgrep/testdata/fixtures/javascript/side-effects.ts
import { exec } from "child_process";
import fs from "fs";

async function f(db, el) {
  // ruleid: js-side-effects
  exec("ls");
  // ruleid: js-side-effects
  fs.writeFileSync("out.txt", "");
  // ruleid: js-side-effects
  await fetch("https://example.com");
  // ruleid: js-side-effects
  db.query("SELECT 1");
  // ruleid: js-side-effects
  el.innerHTML = "<b>hi</b>";
  // ok: js-side-effects
  console.log("hi");
}
